scripts/lib/installer-win32-gen.tsx

```diff
@@ -151,7 +155,91 @@ function getDescendantDirs(d: directory): directory[] {
  */
 export default async function generateFileList(rootPath: string): Promise<string> {
   const rootDir = await walk(rootPath);
-  const descendantDirs = getDescendantDirs(rootDir);
+  const descendantDirs = getDescendantDirs(rootDir).filter(d => d.files.length > 0);
+
+  const specialComponents: Record<string, (d: directory, f: { name: string, id: string }) => Element> = {
+    'Rancher Desktop.exe': (d, f) => {
+      return <Component>
+        <File
+          Name={f.name}
+          Source={path.join('$(var.appDir)', f.name)}
+          ReadOnly="yes"
+          KeyPath="yes"
+          Id="mainExecutable">
+          <Shortcut
+            Id="desktopShortcut"
+            Directory="DesktopFolder"
+            Name="Rancher Desktop"
+            WorkingDirectory="APPLICATIONFOLDER"
+            Advertise="yes"
+            Icon="RancherDesktopIcon.exe" />
+          <Shortcut
+            Id="startMenuShortcut"
+            Directory="ProgramMenuFolder"
+            Name="Rancher Desktop"
+            WorkingDirectory="APPLICATIONFOLDER"
+            Advertise="yes"
+            Icon="RancherDesktopIcon.exe">
+            <ShortcutProperty
+              Key="System.AppUserModel.ID"
+              Value="io.rancherdesktop.app" />
+          </Shortcut>
+        </File>
+      </Component>;
+    },
+
+    'resources\\resources\\win32\\internal\\privileged-service.exe': (d, f) => {
+      return <Component>
+        <Condition>NOT MSIINSTALLPERUSER</Condition>
+        <File
+          Name={f.name}
+          Source={path.join('$(var.appDir)', d.name, f.name)}
+          ReadOnly="yes"
+          KeyPath="yes"
+          Id={f.id}
+        />
+        <ServiceInstall
+          DisplayName="Rancher Desktop Privileged Service"
+          ErrorControl="ignore"
+          Name="RancherDesktopPrivilegedService"
+          Start="demand"
+          Type="ownProcess"
+        >
+          {/* SDDL explanation
+            * O:SY  // Owner: SDDL_LOCAL_SYSTEM
+            * D:()  // DACL (see ACE strings)
+            * A;    // ACE type: SDDL_ACCESS_ALLOWED
+            * ;     // ACE flags: none
+            * GRGX; // Rights: GENERIC_READ + GENERIC_EXECUTE
+            * ;     // Object GUID: none
+            * ;     // Inherit Object GUID: none
+            * IU    // Account SID: SDDL_INTERACTIVE
+            *       // Resource attribute: none
+            * And for the second ACE, needed for uninstall:
+            * A;    // ACE type: SDDL_ACCESS_ALLOWED
+            * ;     // ACE flags: none
+            * GA;   // Rights: GENERIC_ALL
+            * ;     // Object GUID: none
+            * ;     // Inherit Object GUID: none
+            * SY    // Accound SID: LOCAL_SYSTEM
+            *       // Resource attribute: none
+            */}
+          <PermissionEx Sddl="O:SYD:(A;;GRGX;;;IU)(A;;GA;;;SY)" />
```

Comment:
The SID string for Interactively logged-on users (`IU`) may not be necessary, and `(A;;GA;;;SY)` might just be adequate. 
Also, we should make sure that the server starts up with this security descriptor.

Reply:
`IU` is needed so that the interactive user can start/stop the service (but can't delete it).

Yes, tested with starting the service (which is why IU is there).